### DISCUSSION TITLE
Dashboard: API connection fails with error 3099

### DISCUSSION THREAD

**Discussion by M-Stenzel** [wazuh/wazuh#26018] [Q&A]
Hi friends,

this is a fresh (no update) Linux installation.

wazuh-indexer-4.9.0-1.x86_64
wazuh-manager-4.9.0-1.x86_64
wazuh-dashboard-4.9.0-2.x86_64

and

filebeat-7.10.2-1.x86_64

I installed according to the documentation (step by step installation).

Unfortunately the API connection fails. 

I followed all the suggestions in 

https://documentation.wazuh.com/current/user-manual/wazuh-dashboard/troubleshooting.html

This is the content of the /etc/wazuh-dashboard/opensearch_dashboards.yml

---

server.host: 10.4.0.2
server.port: 13443
opensearch.hosts: https://10.4.0.2:9200/
opensearch.ssl.verificationMode: certificate
#opensearch.username:
#opensearch.password:
opensearch.requestHeadersAllowlist: ["securitytenant","Authorization"]
opensearch_security.multitenancy.enabled: false
opensearch_security.readonly_mode.roles: ["kibana_read_only"]
server.ssl.enabled: true
server.ssl.key: "/etc/wazuh-dashboard/certs/dashboard-key.pem"
server.ssl.certificate: "/etc/wazuh-dashboard/certs/dashboard.pem"
opensearch.ssl.certificateAuthorities: ["/etc/wazuh-dashboard/certs/root-ca.pem"]
uiSettings.overrides.defaultRoute: /app/wz-home

---

This is the content of /usr/share/wazuh-dashboard/data/wazuh/config/wazuh.yml

---

hosts:
  - default:
    url: https://10.4.0.2/
    port: 55000
    username: wazuh-wui
    password: "redacted"
    run_as: false
---

Could be very important: I had to create the directory /usr/share/wazuh-dashboard/data/wazuh/ and the file

/usr/share/wazuh-dashboard/data/wazuh/config/wazuh.yml. Directory and file did not come with the rpm package.

I checked the API availability

# curl -k -X GET ["https://10.4.0.2:55000/"](https://10.4.0.2:55000/) -H "Authorization: Bearer $(curl -u wazuh-wui:redacted
-k -X POST 'https://10.4.0.2:55000/security/user/authenticate?raw=true')"
 % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                Dload  Upload   Total   Spent    Left  Speed
100   404  100   404    0     0   1065      0 --:--:-- --:--:-- --:--:--  1068
{"data": {"title": "Wazuh API REST", "api_version": "4.9.0", "revision": 40907, "license_name": "GPL 2.0", "license_url": "htt
ps://github.com/wazuh/wazuh/blob/v4.9.0/LICENSE", "hostname": "xxx.xy-space.de", "timestamp": "2024-09-29T14:20:39Z"}, "erro
r": 0}

I checked the running processes

# /var/ossec/bin/wazuh-control status
wazuh-clusterd not running...
wazuh-modulesd is running...
wazuh-monitord is running...
wazuh-logcollector is running...
wazuh-remoted is running...
wazuh-syscheckd is running...
wazuh-analysisd is running...
wazuh-maild not running...
wazuh-execd is running...
wazuh-db is running...
wazuh-authd is running...
wazuh-agentlessd not running...
wazuh-integratord not running...
wazuh-dbd not running...
wazuh-csyslogd not running...
wazuh-apid is running...

Additional results:

# curl https://10.4.0.2:9200/_cat/indices/wazuh-alerts-* -u wazuh-wui:redacted
Q -k

No output from that command.

# curl https://10.4.0.2:9200/_cat/indices/wazuh-* -u wazuh-wui:redacted
Q -k

No output from that command either!

Additional results:

# filebeat test output
elasticsearch: https://10.4.0.2:9200/...
 parse url... OK
 connection...
   parse host... OK
   dns lookup... OK
   addresses: 10.4.0.2
   dial up... OK
 TLS...
   security: server's certificate chain verification is enabled
   handshake... OK
   TLS version: TLSv1.3
   dial up... OK
 talk to server... OK
 version: 7.10.2

Checking the logs...
# cat /var/log/wazuh-indexer/wazuh-cluster.log | grep -i -E "error|warn"

...

[2024-09-29T17:39:54,473][WARN ][o.o.p.c.u.JsonConverter  ] [node-1] Json Mapping Error: Cannot invoke "java.lang.Long.longVal
ue()" because "this.cacheMaxSize" is null (through reference chain: org.opensearch.performanceanalyzer.collectors.CacheConfigM
etricsCollector$CacheMaxSizeStatus["Cache_MaxSize"])
[2024-09-29T17:39:59,473][WARN ][o.o.p.c.u.JsonConverter  ] [node-1] Json Mapping Error: Cannot invoke "java.lang.Long.longVal
ue()" because "this.cacheMaxSize" is null (through reference chain: org.opensearch.performanceanalyzer.collectors.CacheConfigM
etricsCollector$CacheMaxSizeStatus["Cache_MaxSize"])
[2024-09-29T17:40:04,474][WARN ][o.o.p.c.u.JsonConverter  ] [node-1] Json Mapping Error: Cannot invoke "java.lang.Long.longVal
ue()" because "this.cacheMaxSize" is null (through reference chain: org.opensearch.performanceanalyzer.collectors.CacheConfigM
etricsCollector$CacheMaxSizeStatus["Cache_MaxSize"])
[2024-09-29T17:40:09,475][WARN ][o.o.p.c.u.JsonConverter  ] [node-1] Json Mapping Error: Cannot invoke "java.lang.Long.longVal
ue()" because "this.cacheMaxSize" is null (through reference chain: org.opensearch.performanceanalyzer.collectors.CacheConfigM
etricsCollector$CacheMaxSizeStatus["Cache_MaxSize"])

...

In the long run I would like to get rid of that warnings, too... 🙂

cat /var/log/filebeat/filebeat | grep -i -E "error|warn"

No output, no warnings, no errors

# cat /var/ossec/logs/ossec.log | grep -i -E "error|warn"
2024/09/29 00:05:50 wazuh-db: WARNING: After vacuum, the database '000' has become just as fragmented or worse
2024/09/29 02:05:52 wazuh-db: WARNING: After vacuum, the database '000' has become just as fragmented or worse
2024/09/29 04:05:54 wazuh-db: WARNING: After vacuum, the database '000' has become just as fragmented or worse
2024/09/29 06:05:57 wazuh-db: WARNING: After vacuum, the database '000' has become just as fragmented or worse
2024/09/29 08:05:59 wazuh-db: WARNING: After vacuum, the database '000' has become just as fragmented or worse
2024/09/29 10:06:01 wazuh-db: WARNING: After vacuum, the database '000' has become just as fragmented or worse
2024/09/29 12:06:03 wazuh-db: WARNING: After vacuum, the database '000' has become just as fragmented or worse
2024/09/29 14:06:05 wazuh-db: WARNING: After vacuum, the database '000' has become just as fragmented or worse

I cannot put these warnings into a context...

What else can I do?

**Comment by rubencastello90** [comment]
Same issue here on fresh install.

Edit: /usr/share/wazuh-dashboard/data/wazuh/config/wazuh.yml

And use default username and password.

     username: wazuh-wui
     password: wazuh-wui

**Reply by M-Stenzel** [reply]
Hi, 
unfortunately this did not change things to a positive...

Did _you_ did this step?

[https://documentation.wazuh.com/current/installation-guide/wazuh-dashboard/step-by-step.html#securing-your-wazuh-installation](https://documentation.wazuh.com/current/installation-guide/wazuh-dashboard/step-by-step.html#securing-your-wazuh-installation)

**Reply by rubencastello90** [reply]
No, I used Assisted Installation: https://documentation.wazuh.com/current/installation-guide/wazuh-dashboard/installation-assistant.html

Should I do it? No mention on assisted installation about that,..

**Comment by M-Stenzel** [comment]
I guess you should do it, since a "password" such as "wazuh-wui" is not a "password". Better ask someone else, too.

**Comment by huyifu777** [comment]
[2024-09-29T17:39:54,473][WARN ][o.o.p.c.u.JsonConverter ] [node-1] Json Mapping Error: Cannot invoke "java.lang.Long.longVal
ue()" because "this.cacheMaxSize" is null (through reference chain: org.opensearch.performanceanalyzer.collectors.CacheConfigM
etricsCollector$CacheMaxSizeStatus["Cache_MaxSize"])

this problem do you fix?   how？
